compiler/quilt/tools/command.py

```diff
@@ -1326,17 +1332,85 @@ def iter_filename_map(node):
         # Technically successful, but with nothing to do.
         # package may have no file nodes, or user may have filtered out all applicable targets.
         # -- should we consider it an error and raise?
-        print("No files to export.")
+        if not silent:
+            print("No files to export.")
         return
 
+    # ensure output path is writable.  I'd just check stat, but this is fully portable.
+    try:
+        output_path.mkdir(exist_ok=True)  # could be '.'
+        with tempfile.TemporaryFile(dir=str(output_path), prefix="quilt-export-write-test-", suffix='.tmp'):
+            pass
+    except OSError as error:
+        raise CommandException("Invalid export path: not writable: " + str(error))
+
     # Paths verified, let's export..
-    sys.stdout.write('Exporting.')
-    sys.stdout.flush()
-    for src, dest in final_export_map:
-        if not dest.parent.exists():
-            dest.parent.mkdir(parents=True, exist_ok=True)
-        sys.stdout.write('.')
-        sys.stdout.flush()
-        dest.touch()  # weird issue: zero-byte files not getting copied?!  TODO: performance
-        copy(str(src), str(dest))
-    print('..done.')
+    try:
+        if not silent:
+            sys.stdout.write('Exporting.')
+            sys.stdout.flush()
+        for src, dest in final_export_map:
+            if not dest.parent.exists():
+                dest.parent.mkdir(parents=True, exist_ok=True)
+            if not silent:
+                sys.stdout.write('.')
+                sys.stdout.flush()
+            if src in zero_byte_files:
+                dest.touch()  # weird issue: zero-byte files not getting copied?!
+            else:
+                copy(str(src), str(dest))
+        if not silent:
+            print('..done.')
+    except OSError as error:
+        commandex = CommandException("Unexpected error during export: " + str(error))
+        commandex.original_error = error
+        raise commandex
+
+
+def _repair_package(packagename):
+    """
```

Comment:
Currently we're not handling locally data store upgrades and instead asking users to clear between upgrades.  This keeps our code cleaner.

If you disagree, pls ping Kevin fire approval

Reply:
For approval

Reply:
No problem -- the _set bug wasn't released anyways, as nothing used _set before this tensorflow branch -- it was just in case your pkgstore was messed up and you didn't want to delete it.